### ISSUE TITLE
Reproduced CAV-MAE Shows Lower Visual Modality Performance — Any Advice?

### ISSUE THREAD

**Issue by Ash-one** [Bizilizi/VGGSounder#3]
Thanks for testing these audio-visual models!

I have a question regarding the inference modes of the audio-visual embedding models. For example, the CAV-MAE fine-tuned model supports multiple modes such as "multimodal", "audioonly", "videoonly", "missingaudioonly", "missingvideoonly".

How did you ensure a fair and comparable evaluation across these different modes?

I reproduced CAV-MAE and fine-tuned it on my VGGSound dataset. For evaluation, I used one video frame per sample together with 10 seconds of audio, and obtained the audio, video, and audio-visual logits using the "missingaudioonly", "missingvideoonly", and "multimodal" modes. However, my results differ from those reported in the paper — especially for the visual-only modality, where the result appears lower across all metrics. 

(As a side note: my evaluation split is missing 15 videos.)

Could you please provide some guidance or suggestions on how to properly run a comparable evaluation?

Thanks in advance!

<html xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:x="urn:schemas-microsoft-com:office:excel" xmlns="http://www.w3.org/TR/REC-html40">
<head>

<meta name=Generator content="Microsoft Excel">

</head>
<body>

metric | accuracy | accuracy | accuracy | f1 | f1 | f1 | f1 | f1 | hit_rate | hit_rate | hit_rate | mu | mu | mu
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
modality | a | v | av | a | v | av | a only | v only | a | v | av | a | v | av
CAV-MAE(Paper) | 12.73 | 19.22 | 24.38 | 33.80 | 34.32 | 41.98 | 14.93 | 19.68 | 61.06 | 51.91 | 62.44 | 4.34 | 5.92 | 0.89 
CAV-MAE(mine) | 12.38 | 11.36 | 24.28 | 32.79 | 21.17 | 41.58 | 14.95 | 13.61 | 59.24 | 32.02 | 61.86 | 3.74 | 3.32 | 0.25

</body>

</html>
